storage/mountscan/mountscan.go

```diff
@@ -0,0 +1,206 @@
+package mountscan
+
+import (
+	"io/ioutil"
+	"strconv"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/contiv/errored"
+	"github.com/contiv/volplugin/errors"
+)
+
+const (
+	mountInfoFile           = "/proc/self/mountinfo"
+	deviceInfoFile          = "/proc/devices"
+	nfsMajorID              = 0
+	totalMountInfoFieldsNum = 11
+)
+
+// GetMountsRequest captures all the params required for scanning mountinfo
+type GetMountsRequest struct {
+	DriverName   string // ceph, nfs
+	FsType       string // nfs4, ext4
+	KernelDriver string // rbd, device-mapper, etc.
```

Comment:
Seems good, my knee-jerk reaction was at the fact we don't need this for anybody but ceph, but the code itself is generic enough so that when something does come along we'll be good.